src/lib.rs

```diff
@@ -987,21 +979,21 @@ impl Drop for Rav1dContext {
                 n_1 = n_1.wrapping_add(1);
             }
             rav1d_free_aligned(self.fc as *mut c_void);
-            if self.n_fc > 1 as c_uint && !(self.frame_thread.out_delayed).is_null() {
+            if self.n_fc > 1 as c_uint && !(self.frame_thread.out_delayed).is_empty() {
                 let mut n_2: c_uint = 0 as c_int as c_uint;
                 while n_2 < self.n_fc {
-                    if (*(self.frame_thread.out_delayed).offset(n_2 as isize))
+                    if (self.frame_thread.out_delayed[n_2 as usize])
                         .p
                         .frame_hdr
                         .is_some()
                     {
                         rav1d_thread_picture_unref(
-                            &mut *(self.frame_thread.out_delayed).offset(n_2 as isize),
+                            &mut self.frame_thread.out_delayed[n_2 as usize],
                         );
                     }
                     n_2 = n_2.wrapping_add(1);
                 }
-                free(self.frame_thread.out_delayed as *mut c_void);
+                let _ = std::mem::take(&mut self.frame_thread.out_delayed);
```

Comment:
```suggestion
                let _ = mem::take(&mut self.frame_thread.out_delayed);
```

We've generally been importing things like `std::mem`, `std::iter`, etc.